pkg/auth/providers/publicapi/login.go

```diff
@@ -237,14 +237,14 @@ func (h *loginHandler) createLoginToken(request *types.APIContext) (v3.Token, st
 		userExtraInfo := providers.GetUserExtraAttributes(providerName, userPrincipal)
 		err = h.tokenMGR.UserAttributeCreateOrUpdate(currUser.Name, userPrincipal.Provider, groupPrincipals, userExtraInfo, loginTime)
 		if err != nil {
-			logrus.Warnf("Error creating or updating userAttribute for %s, retrying: %v", currUser.Name, err)
+			logrus.Warnf("Error creating or updating userAttribute for %s, retrying: %v", userPrincipal.Name, err)
```

Comment:
What about `currUser` in `UserAttributeCreateOrUpdate(currUser.Name...` just above the condition ?

Reply:
It's unaffected. This call is unreachable if `EnsureUser()` returns an error.

Reply:
ok

Reply:
The sources of potential panics are L227 and L247